### PR TITLE
Add comments about initialisms and resource names

### PR DESCRIPTION
**Pull request by tnqn**
Co-authored-by: Jianjun Shen <shenj@vmware.com>
Signed-off-by: Quan Tian <qtian@vmware.com>

### CODE REVIEW

**Comment by tnqn** [comment]
cc @jianjuns

For "Antrea Controller or `antrea-controller` for component names, with the latter referring to binary (daemon/container) names", I wonder how we should write the former case of "antctl", "kube-proxy"?

**Comment by jianjuns** [comment]
Do you mean you like to add more examples like anctl and kube-proxy? That sounds good to me.

BTW, `` should be for documentation only. Not needed in code.

**Comment by tnqn** [comment]
> Do you mean you like to add more examples like anctl and kube-proxy? That sounds good to me.

No, I meant I don't know what's the proper writing of anctl and kube-proxy, corresponding to "Antrea Controller", if we add this suggestion. It seems it only applies to antrea-agent and antrea-controller.

**Comment by jianjuns** [comment]
They should just be anctl and kube-proxy.  I think the difference is most our components are named with a meaningful name like Antrea Controller, Antrea Multi-cluster Controller, etc., while kube-proxy is not like that (I do see K8s documentation calls it Kubernetes Proxy in a few cases though, but that is not well known to people).

File: README.md [hunk 1]
```diff
@@ -108,6 +110,42 @@ type RouteClient struct {
...
+// TCPHeader contains the TCP specific header.
+type TCPHeader struct { ... }
+
+func (p *Packet) SetTCPHeader(tcp *TCPHeader) { ... }
+```
+
+## Use upper camel case for K8s resource and CRD kind names
+
+In comments and documentation, use upper camel case for K8s resource and CRD
+kind names, e.g. "Pod", "Service", "TrafficControl".
```
Comment (jianjuns) [inline README.md]: Could we add Node and NetworkPolicy too. They are used a lot in Antrea, and people often use different formats.

Comment (tnqn) [reply README.md]: added

**Comment by tnqn** [comment]
> They should just be anctl and kube-proxy. I think the difference is most our components are named with a meaningful name like Antrea Controller, Antrea Multi-cluster Controller, etc., while kube-proxy is not like that (I do see K8s documentation calls it Kubernetes Proxy in a few cases though, but that is not well known to people).

added, PTAL

File: README.md [hunk 2]
```diff
@@ -108,6 +110,59 @@ type RouteClient struct {
...
+type TCPHeader struct { ... }
+
+func (p *Packet) SetTCPHeader(tcp *TCPHeader) { ... }
+```
+
+## Write proper nouns formally
+
+In comments and documentation, use **upper camel case** for K8s resource and CRD
+kind names, e.g. "Pod", "Node", "Service", "NetworkPolicy", "TrafficControl".
+For component names, **capitalize the first letters** if a component name
```
Comment (jianjuns) [inline README.md:141]: Should we start a separate paragraph after the resource / CRD examples?

**Review by jianjuns — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (tnqn) [reply README.md:141]: done

**Comment by jianjuns** [comment]
LGTM.

**Pull request merged**